src/entities/Coauthor/routes.ts

```diff
@@ -16,16 +20,35 @@ export default routes((route) => {
   route.put('/coauthors/:proposal', withAuth, handleAPI(updateStatus))
 })
 
+async function filterCoauthorRequests(requests: CoauthorAttributes[]) {
+  const now = Date.now()
+  const filteredRequests: CoauthorAttributes[] = []
+  for (const request of requests) {
+    if (request.status === CoauthorStatus.PENDING) {
+      const proposal = await ProposalModel.findOne<ProposalAttributes>({ id: request.proposal_id, deleted: false })
+      if (proposal && now < proposal.finish_at.getTime()) {
```

Comment:
We are repeating this check of the proposal date here and on the `updateStatus` function: 
`proposal && now < proposal.finish_at.getTime()`
`Date.now() < proposal.finish_at.getTime()`

(also in the front `const isUpdatable = Date.now() < proposalFinishDate.getTime()`

It's a simple logic but I wonder if we should encapsulate it in a function, at least to make the check more explicit in its intention.

Maybe in proposal utils:

`function coauthoringIsUpdatable(whatevs)`